mautic/mautic
# Not store ip if anonymization is on

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://github.com/mautic/mautic/issues/12671 




#### Description:

Problem -
Anonymize Ip feature was already exist based on configuration, it was anonymizing IP but As per GDPR rule we will not store Ip address.

Solution

Anonymize existing IP addresses when the IP anonymization is turned on.
Warn the user that IP addresses will be anonymized. Show a big red box with a warning next to the "Anonymize IP" config option. Use the following warning text: "Existing data will be anonymized. This action is not reversible".
Do not store IP address if the feature is enabled


#### Steps to Reproduce:

1. Turn on Anonymize IP feature from Configuration -> Tracking Settings -> Mautic tracking settings
2. Now After page visit check `ip_addresses` table, Ip address are being store but those are anonymize.
 
#### Steps to test this PR:

1. Turn on Anonymize IP feature from Configuration -> Tracking Settings -> Mautic tracking settings
2. All existing Ip address should be anoymize from ip_addresses table.
3. In Anonymous contact list, IP address should not be visible for existing anonymous contacts.
4. Visit any landing page, download assets , submit form etc. but IP address should not be stored in ip_addresses table.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10116"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
